### PR TITLE
Various Dutch address and name improvements.

### PR DESCRIPTION
**Pull request by bodiam**
Based on: https://onzetaal.nl/taalloket/afkortingen-van-provincienamen

### CODE REVIEW

File: src/test/java/net/datafaker/providers/base/AddressTest.java [hunk 1]
```diff
@@ -261,4 +261,10 @@ void nonDefaultLocaleStreetName(String locale) {
         BaseFaker localFaker = new BaseFaker(new Locale(locale));
         assertThat(localFaker.address().streetName()).isNotEmpty();
     }
+
+    @RepeatedTest(100)
+    void dutchAddress() {
+        assertThat(NL_FAKER.address().stateAbbr()).matches("[A-Z]{2}");
+        assertThat(NL_FAKER.address().fullAddress()).matches("[A-Z].+, [0-9]{4} [A-Z]{2}, [A-Z].+");
```
Comment (snuyanzin) [inline src/test/java/net/datafaker/providers/base/AddressTest.java:268]: should it check for only uppercase?

Comment (bodiam) [reply src/test/java/net/datafaker/providers/base/AddressTest.java:268]: it should check that the fullAddress starts with an uppercase, followed by anything.

Comment (bodiam) [reply src/test/java/net/datafaker/providers/base/AddressTest.java:268]: (I've ran this test approx 1_000_000 times, but I thought that was a bit too much to commit)

Comment (snuyanzin) [reply src/test/java/net/datafaker/providers/base/AddressTest.java:268]: ok, i see, thanks

**Review by snuyanzin — APPROVED** [review]

**Pull request merged**